src/modules/DetectionSensorModule.cpp

```diff
@@ -0,0 +1,94 @@
+#include "DetectionSensorModule.h"
+#include "MeshService.h"
+#include "NodeDB.h"
+#include "PowerFSM.h"
+#include "configuration.h"
+#include "main.h"
+
+DetectionSensorModule *detectionSensorModule;
+
+#define GPIO_POLLING_INTERVAL 80
+#define DELAYED_INTERVAL 1000
+
+int32_t DetectionSensorModule::runOnce()
+{
+    /*
+        Uncomment the preferences below if you want to use the module
+        without having to configure it from the PythonAPI or WebUI.
+    */
+    // moduleConfig.detection_sensor.enabled = true;
+    // moduleConfig.detection_sensor.monitor_pin = 10; // WisBlock PIR IO6
+    // moduleConfig.detection_sensor.minimum_broadcast_secs = 60;
+    // moduleConfig.detection_sensor.state_broadcast_secs = 120;
+    // moduleConfig.detection_sensor.detection_triggered_high = false;
+    // strcpy(moduleConfig.detection_sensor.name, "Motion");
+
+    if (moduleConfig.detection_sensor.enabled == false)
+        return disable();
+
+    if (firstTime) {
+        // This is the first time the OSThread library has called this function, so do some setup
+        firstTime = false;
+        if (moduleConfig.detection_sensor.monitor_pin > 0) {
```

Comment:
What if the monitor_pin is kept 0 and the module is enabled? Looks like hasStateChanged() tries to read from GPIO0.